### PR TITLE
Auto-create PSResourceRepository.xml file if the file does not already exist. 

### LINKED ISSUE #258: First run experience is... lacking. Cannot find XML
Testing PowerShellGet 3.0.0-beta10 with CompatPowerShellGet 0.0.2, on both PowerShell 7.0.3 and 7.1.0-preview.7.

First thing I did with the new module was try and install the SecretManagement module.

```powershell
Install-Module -Name Microsoft.PowerShell.SecretManagement -AllowPrerelease
```

I'm told Install-Module is deprecated (expected), but then get an error about not finding XML and being told to register a repo.

```output
WARNING: The cmdlet 'Install-Module' is deprecated, please use 'Install-PSResource'.
Install-PSResource: Was not able to successfully find xml. Try running 'Register-PSResourceRepository -PSGallery'
```

It seems that this error occurs when running *any* PowerShellGet command, including `Get-PSResourceRepository`.

As a first run experience with the module and expecting it to "just work", this is jarring. Surley PSGallery should be a default (as I'm sure most users will be using that, and only that... advanced users should be more than capable of removing it if needed?)

### PR DESCRIPTION
**Pull request by alerickson**
# PR Summary
Automatically create the PSGet repository store (PSResourceRepository.xml file) if the file does not already exist. 
This functionality already existed in the *-PSResourceRepository cmdlets, but did not exist for the *-PSResource cmdlets.  All of the latter cmdlets that read from the PSResourceRepository.xml file (ie, Find, Install, Save, and Publish) now check to makes sure that the file already exists before processing.  If the file does not exist, it gets created.

Another change is that upon each new PSResourceRepository.xml file creation, the PowerShellGallery gets registered by default with the following settings:
```
Name:  PSGallery
URL:  https://powershellgallery.com/api/v2
Priority: 50
Trusted: false
```

## PR Context

Resolves #258 

## PR Checklist

- [x] [PR has a meaningful title](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
    - Use the present tense and imperative mood when describing your changes
- [x] [Summarized changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [x] [Make sure all `.h`, `.cpp`, `.cs`, `.ps1` and `.psm1` files have the correct copyright header](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [x] This PR is ready to merge and is not [Work in Progress](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---work-in-progress).
    - If the PR is work in progress, please add the prefix `WIP:` or `[ WIP ]` to the beginning of the title (the `WIP` bot will keep its status check at `Pending` while the prefix is present) and remove the prefix when the PR is ready.
- **[Breaking changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#making-breaking-changes)**
    - [x] None
    - **OR**
    - [ ] [Documentation needed]()
        - [ ] Issue filed: 
- **User-facing changes**
    - [x] Not Applicable
    - **OR**
    - [ ] [Documentation needed]()
        - [ ] Issue filed: 
- **Testing - New and feature**
    - [x] N/A or can only be tested interactively
    - **OR**
    - [ ] [Make sure you've added a new test if existing tests do not effectively test the code changed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#before-submitting)
- **Tooling**
    - [x] I have considered the user experience from a tooling perspective and don't believe tooling will be impacted.
    - **OR**
    - [ ] I have considered the user experience from a tooling perspective and enumerated concerns in the summary.

### CODE REVIEW

File: src/code/RepositorySettings.cs [hunk 1]
```diff
@@ -27,7 +27,10 @@ internal static class RepositorySettings
         private static readonly string RepositoryFileName = "PSResourceRepository.xml";
         private static readonly string RepositoryPath = Path.Combine(Environment.GetFolderPath(SpecialFolder.LocalApplicationData), "PowerShellGet");
         private static readonly string FullRepositoryPath = Path.Combine(RepositoryPath, RepositoryFileName);
-
+        private static readonly string PSGalleryRepoName = "PSGallery";
```
Comment (PaulHigin) [inline src/code/RepositorySettings.cs]: These should be 'private const string' types.

File: src/code/RepositorySettings.cs [hunk 2]
```diff
@@ -27,7 +27,10 @@ internal static class RepositorySettings
         private static readonly string RepositoryFileName = "PSResourceRepository.xml";
```
Comment (PaulHigin) [inline src/code/RepositorySettings.cs]: This should also be 'private const string'

File: src/code/FindPSResource.cs [hunk 3]
```diff
@@ -134,7 +134,22 @@ public sealed class FindPSResource : PSCmdlet
...
-            _cancellationToken = _source.Token;
+            _cancellationToken = _source.Token;
+
+            // Create a respository story (the PSResourceRepository.xml file) if it does not already exist
+            // This is to create a better experience for those who have just installed v3 and want to get up and running quickly
+            try
+            {
+                RepositorySettings.CheckRepositoryStore();
+            }
+            catch (PSInvalidOperationException e)
```
Comment (PaulHigin) [inline src/code/FindPSResource.cs]: It feels unnecessary to catch and rethrow the PSInvalidOperationException error.  I think we can simplify this by removing the try/catch and just have 'RepositorySettings.CheckRepositoryStore();' check, and let the original thrown exception terminate the cmdlet.  The error message is clear, and the error stack is preserved.

**Review by PaulHigin — CHANGES_REQUESTED** [review]

**Review by PaulHigin — APPROVED** [review]

File: src/code/FindPSResource.cs (hunk 3 above)
Comment (anamnavi) [reply src/code/FindPSResource.cs]: I agree.

**Review by anamnavi — APPROVED** [review]
LGTM

**Pull request merged**